spring-integration-cloudevents/src/main/java/org/springframework/integration/cloudevents/transformer/ToCloudEventsTransformer.java

```diff
@@ -0,0 +1,323 @@
+	 * Defaults to {@code "/spring/" + appName + "." + getBeanName())}.
+	 * @param sourceExpression the expression to create the source for each CloudEvent
+	 */
+	public void setSourceExpression(Expression sourceExpression) {
+		this.sourceExpression = sourceExpression;
+	}
+
+	/**
+	 * Set the {@link Expression} for extracting the type for the CloudEvent.
+	 * Defaults to "spring.message".
+	 * @param typeExpression the expression to create the type for each CloudEvent
+	 */
+	public void setTypeExpression(Expression typeExpression) {
+		this.typeExpression = typeExpression;
+	}
+
+	/**
+	 * Set the {@link Expression} for creating the dataSchema for the CloudEvent.
+	 * Defaults to null.
+	 * @param dataSchemaExpression the expression to create the dataSchema for each CloudEvent
+	 */
+	public void setDataSchemaExpression(@Nullable Expression dataSchemaExpression) {
+		this.dataSchemaExpression = dataSchemaExpression;
+	}
+
+	/**
+	 * Set the {@link Expression} for creating the subject for the CloudEvent.
+	 * Defaults to null.
+	 * @param subjectExpression the expression to create the subject for each CloudEvent
+	 */
+	public void setSubjectExpression(@Nullable Expression subjectExpression) {
+		this.subjectExpression = subjectExpression;
+	}
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+		ApplicationContext applicationContext = getApplicationContext();
+		if (this.sourceExpression == null) {  // in the case the user sets the value prior to onInit.
+			String appName = applicationContext.getEnvironment().getProperty("spring.application.name");
+			appName = appName == null ? "unknown" : appName;
+			this.sourceExpression = new ValueExpression<>(URI.create("/spring/" + appName + "." + getBeanName()));
+		}
+		this.cloudEventMessageConverter = new CloudEventMessageConverter(this.cloudEventPrefix, this.getSpecVersionKey(), this.getDataContentTypeKey());
+	}
+
+	/**
+	 * Transform the input message into a CloudEvent message.
+	 * @param message the input Spring Integration message to transform
+	 * @return CloudEvent message in the specified format
+	 * @throws RuntimeException if serialization fails
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	protected Object doTransform(Message<?> message) {
+		Assert.isInstanceOf(byte[].class, message.getPayload(), "Message payload must be of type byte[]");
+
+		String id = this.eventIdExpression.getValue(this.evaluationContext, message, String.class);
+		URI source = this.sourceExpression.getValue(this.evaluationContext, message, URI.class);
+		String type = this.typeExpression.getValue(this.evaluationContext, message, String.class);
+
+		String contentType = message.getHeaders().get(MessageHeaders.CONTENT_TYPE, String.class);
+		if (contentType == null) {
+			throw new MessageTransformationException(message, "Missing 'Content-Type' header");
```

Comment:
I think we should fallback to the `application/octet-stream` since we have already checked that payload is a `byte[]`.